lhotse/cut.py

```diff
@@ -1132,6 +1132,26 @@ def describe(self) -> None:
         with pd.option_context('precision', 1):
             print(durations.describe().drop('count'))
 
+    def add_supervision(
```

Comment:
I'm trying to figure out what is the motivation for this method. Am I correct in guessing that it allows you to replace the original supervisions with different ones from another source (e.g. ground truth -> forced alignment or predictions) and get a new CutSet? I'm not sure if this is the right way to do this, since these supervisions could actually exceed the bounds of the cut. I think I'd rather create a new CutSet using the updated SupervisionSet and the original RecordingSet. Does that make sense for your use-case?

Reply:
Okay yeah that makes sense. I was using `trim_to_supervisions()` after adding the new supervisions which probably circumvented the bounds problem. But it's better to create a new CutSet altogether.

Reply:
I meant cases where a different SupervisionSet implies re-segmentation, e.g. segment 5s - 10s changed into a segment of 5.2s to 10.5s - then it's mismatched with the cut (which would have matched the "old" segments boundaries when created with `trim_to_supervisions`.